lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -105,22 +105,25 @@ LinearLayout makeCgaLayout(CTALayoutAttr layout) {
   return ret.transposeOuts(outDimNames);
 }
 
-// Shrinks the output set of a layout function while leaving the input set
-// unchanged, by making high-order inputs in inDimName map to the same output.
-// Attempts to shrink down to desiredSize, but this is not always possible just
-// by modifying one the specified input dimension.
+// For each out-dim d, ensure the layout's out-size (i.e. its codomain) is no
+// larger than shape[d].  Do this without changing the size of the layout's
+// inputs (i.e. leave its domain unchanged).
+//
+// This function is invariant to the order of the layout's input and output
+// dimensions.
 //
-// We do this by making the most-major inputs to the layout map to 0.  This
-// effectively duplicates data along that input dimension.  For example, this
-// layout has out-dim size 32:
+// We do this by making the largest value each output dimension 0 because
+// bases that map to a location larger than the input dimension is essentially
+// duplicating along that dimension.
+// For example, this layout has out-dim size 32, and suppose the shape is [8]:
```

Comment:
> and suppose the shape is [8]:

What shape is [8]?  The input shape of this layout is [register=8, lane=4] and the output shape of the layout is [32].

I think you mean

> For example, this layout has out-dim size 32.  Suppose we call ensureLayoutNotLargerThan to shrink its out-dim to 8.